beacon_node/beacon_chain/src/beacon_chain.rs

```diff
@@ -5543,28 +5548,35 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
             let (mut state, state_root) = if let Some((state, state_root)) = head_state_opt {
                 (state, state_root)
             } else {
-                let state_root = head_block.state_root;
+                // There's no reason we'd want to load the shuffling for a cold
+                // state; attestation duties will never be pre-finalized and
+                // attestations to finalized blocks are useless.
+                let split_slot = self.store.get_split_slot();
+                if split_slot > target_block.slot {
```

Comment:
Double-check that the split slot is the first hot state not the last cold state.

Reply:
Yeah that's right

> Split Slot: the slot at which states are divided between the hot and the cold DBs. All states from slots less than the split slot are in the freezer, while all states with slots greater than or equal to the split slot are in the hot DB.

From https://lighthouse-book.sigmaprime.io/advanced_database.html#glossary